jobs/kpi-forecasting/kpi-forecasting/models/prophet_forecast.py

```diff
@@ -0,0 +1,119 @@
+import json
+import pandas as pd
+import prophet
+
+from datetime import datetime
+from dataclasses import dataclass
+from models.base_forecast import BaseForecast
+from typing import Dict
+
+
+@dataclass
+class ProphetForecast(BaseForecast):
+    @property
+    def column_names_map(self) -> Dict[str, str]:
+        return {"submission_date": "ds", "value": "y"}
+
+    def _fit(self) -> None:
+        """
+        Fit a Prophet model using the `observed_df` that was generated using
+        Metric Hub. This method updates `self.model`.
+        """
+        self.model = prophet.Prophet(
+            **self.parameters,
+            uncertainty_samples=self.number_of_simulations,
+            mcmc_samples=0,
+        )
+
+        if self.use_holidays:
+            self.model.add_country_holidays(country_name="US")
+
+        # Modify observed data to have column names that Prophet expects, and fit
+        # the model
+        self.model.fit(self.observed_df.rename(columns=self.column_names_map))
+
+    def _predict(self) -> pd.DataFrame:
+        """
+        Forecast using `self.model`.
```

Comment:
Is this comment providing much value to the readers understanding?

Reply:
Removed.